src/ralph/cli.py

```diff
@@ -0,0 +1,124 @@
+"""Ralph CLI entrypoint"""
+
+import logging
+import sys
+from inspect import signature
+
+import click
+import click_log
+from click_option_group import optgroup
+
+from ralph.defaults import (
+    AVAILABLE_PARSERS,
+    AVAILABLE_STORAGE_BACKENDS,
+    ENVVAR_PREFIX,
+    Parsers,
+    StorageBackends,
+)
+from ralph.utils import get_class_from_name, get_instance_from_class, get_root_logger
+
+# cli module logger
+logger = logging.getLogger(__name__)
+click_log.basic_config(logger)
+
+
+PARSERS = list(AVAILABLE_PARSERS)
+STORAGE_BACKENDS = list(AVAILABLE_STORAGE_BACKENDS)
+
+
+@click.group(name="ralph")
+@click_log.simple_verbosity_option(get_root_logger())
+def cli():
+    """Ralph core CLI"""
+
+
+def backends_options(name=None, backends=None):
+    """Backend-related options decorator for Ralph commands"""
+
+    def wrapper(command):
+        command = (
+            click.option(
+                "-b",
+                "--backend",
+                type=click.Choice(backends),
+                required=True,
+                help="Storage backend",
+            )
+        )(command)
+
+        for backend in backends:
+            backend_class = get_class_from_name(backend, StorageBackends)
+
+            for parameter in signature(backend_class.__init__).parameters.values():
+                if parameter.name == "self":
+                    continue
+                option = f"--{backend_class.name}-{parameter.name}".replace("_", "-")
+                envvar = (
+                    f"{ENVVAR_PREFIX}_{backend_class.name}_{parameter.name}".upper()
+                )
+                command = (optgroup.option(option, envvar=envvar))(command)
+            command = (optgroup.group(f"{backend_class.name} storage backend"))(command)
+
+        command = (cli.command(name=name or command.__name__))(command)
+        return command
+
+    return wrapper
+
+
+@cli.command()
+@click.option(
+    "-p",
+    "--parser",
+    type=click.Choice(PARSERS),
+    required=True,
+    help="Container format parser used to extract events",
+)
+@click.option(
+    "-c",
+    "--chunksize",
+    type=int,
+    default=50000,
```

Comment:
According to the docstring in the GELFParser.parse method, the recommend benchmarked chunksize is below 10k. Maybe this default should be adjusted?

Reply:
I've added a `DEFAULT_GELF_PARSER_CHUNCK_SIZE` variable for this.